rest-catalog/quarkus-rest-catalog/src/main/java/org/projectnessie/restcatalog/server/auth/OAuthUtils.java

```diff
@@ -0,0 +1,119 @@
+
+import static org.apache.iceberg.rest.auth.OAuth2Util.parseScope;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.vertx.core.json.JsonObject;
+import io.vertx.mutiny.core.buffer.Buffer;
+import io.vertx.mutiny.ext.web.client.HttpResponse;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.iceberg.rest.RESTUtil;
+import org.apache.iceberg.rest.auth.OAuth2Properties;
+import org.apache.iceberg.rest.responses.OAuthTokenResponse;
+import org.projectnessie.restcatalog.api.errors.OAuthTokenEndpointException;
+import org.projectnessie.restcatalog.service.auth.OAuthTokenRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class OAuthUtils {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(OAuthUtils.class);
+
+  private static final String ACCESS_TOKEN = "access_token";
+  private static final String TOKEN_TYPE = "token_type";
+  private static final String EXPIRES_IN = "expires_in";
+  private static final String ISSUED_TOKEN_TYPE = "issued_token_type";
+  private static final String SCOPE = "scope";
+  private static final String ERROR = "error";
+  private static final String ERROR_DESCRIPTION = "error_description";
+
+  /**
+   * Similar to {@link
+   * org.apache.iceberg.rest.auth.OAuth2Util#tokenResponseToJson(OAuthTokenResponse)}.
+   */
+  public static OAuthTokenResponse tokenResponseFromJson(JsonObject json) {
+    OAuthTokenResponse.Builder builder =
+        OAuthTokenResponse.builder()
+            .withToken(json.getString(ACCESS_TOKEN))
+            .withTokenType(json.getString(TOKEN_TYPE));
+    if (json.containsKey(ISSUED_TOKEN_TYPE)) {
+      builder.withIssuedTokenType(json.getString(ISSUED_TOKEN_TYPE));
+    } else {
+      // Iceberg won't be able to refresh tokens without a value for this field,
+      // se we need to set it to something that will work.
+      builder.withIssuedTokenType(OAuth2Properties.ACCESS_TOKEN_TYPE);
+    }
+    if (json.containsKey(EXPIRES_IN)) {
+      builder.setExpirationInSeconds(json.getInteger(EXPIRES_IN));
+    }
+    if (json.containsKey(SCOPE)) {
+      builder.addScopes(parseScope(json.getString(SCOPE)));
+    }
+    return builder.build();
+  }
+
+  /**
+   * Similar to {@link org.apache.iceberg.rest.responses.OAuthErrorResponseParser#fromJson(int,
+   * JsonNode)}.
+   */
+  public static OAuthTokenEndpointException errorFromJson(JsonObject json, int statusCode) {
+    String error = json.getString(ERROR);
+    String errorDescription = json.getString(ERROR_DESCRIPTION);
+    if (error == null) {
+      error = "UnknownOAuthTokenEndpointError";
+      errorDescription = json.encode();
+    }
+    return new OAuthTokenEndpointException(statusCode, error, errorDescription);
+  }
+
+  public static void printOAuthTokenRequest(OAuthTokenRequest req) {
```

Comment:
Isn't it unsafe to dump sensitive information?

Reply:
Yes this is VERY unsafe :-D This was useful for me while developing this PR but I think we should simply get rid of it now.